virttest/qemu_devices/qcontainer.py

```diff
@@ -1569,12 +1569,13 @@ def images_define_by_params(self, name, image_params, media=None,
         shared_dir = os.path.join(data_dir.get_data_dir(), "shared")
         drive_format = image_params.get("drive_format")
         scsi_hba = image_params.get("scsi_hba")
-        if drive_format == "virtio":    # translate virtio to ccw/device
-            machine_type = image_params.get("machine_type")
-            if "s390" in machine_type:      # s390
+        machine_type = image_params.get("machine_type")
+        if "s390" in machine_type:      # s390
+            scsi_hba = "virtio-scsi-ccw"
+            if drive_format == "virtio":
                 drive_format = "virtio-blk-ccw"
-            elif "mmio" in machine_type:    # mmio-based machine
-                drive_format = "virtio-blk-device"
+        elif "mmio" in machine_type:    # mmio-based machine
+            drive_format = "virtio-blk-device"
```

Comment:
Before, for `mmio`, it  only use `drive_format = "virtio-blk-device"` if `drive_format == "virtio"` .  and the new change remove it, this is not a correct logic.

Reply:
ok，keep the logic,and add right scsi_hba to another line.thanks for your comment.